File: maximum-length-of-pair-chain/maximum-length-of-pair-chain.cpp
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        int n = pairs.size(), sol = 1;
        vector<int> dp (n, 1);
        sort(pairs.begin(), pairs.end(), [] 
        (const vector<int>& a, const vector<int>& b) {
            return a[0] < b[0];
        });

        for (int i = n - 2; i >= 0; i--) {
            for (int j = i + 1; j < n; j++) {
                if (pairs[i][1] < pairs[j][0]) {
                    dp[i] = max(dp[i], 1 + dp[j]);
                }//if
            } //for
            sol = max(sol, dp[i]);
        } //for

        return sol;
    } //findLongest
};

File: LeetCode/646-maximum-length-of-pair-chain/maximum-length-of-pair-chain.java
class Solution {
    public int findLongestChain(int[][] pairs) {
        Arrays.<int []>sort(pairs, (a, b) -> (Integer.compare(a[1], b[1])));
        int sol = 1, end = pairs[0][1];

        for (int i = 1; i < pairs.length; i++) {
            if (pairs[i][0] > end) {
                //System.out.println(pairs[i][0] + "," + pairs[i][1]);
                ++sol;
                end = pairs[i][1];
            } //if
        } //for
        
        return sol;
    } //findLongest
} //Sol